evalml/tests/automl_tests/test_automl_search_classification.py

```diff
@@ -606,3 +611,18 @@ def test_automl_allowed_pipelines_search(mock_fit, mock_score, dummy_binary_pipe
     assert start_iteration_callback.call_count == 2
     assert start_iteration_callback.call_args_list[0][0][0] == ModeBaselineBinaryPipeline
     assert start_iteration_callback.call_args_list[1][0][0] == dummy_binary_pipeline_class
+
+
+def test_categorical_hyperparam(X_y_multi):
+    X, y = X_y_multi
+
+    class CustomPipeline(MulticlassClassificationPipeline):
+        component_graph = ['Imputer', 'One Hot Encoder', 'Standard Scaler', 'Logistic Regression Classifier']
+        custom_hyperparameters = {
+            'Simple Imputer': {
+                'impute_strategy': Categorical(['mean', 'most_frequent'])
+            }
+        }
+
+    automl = AutoMLSearch(problem_type="multiclass", allowed_pipelines=[CustomPipeline])
+    automl.search(X, y)
```

Comment:
@bchen1116 just a tip, for tests which don't check scoring, we usually mock fit/score so that we save time in the unit test. See some of the other tests in this file
```python
@patch('evalml.pipelines.BinaryClassificationPipeline.score')
@patch('evalml.pipelines.BinaryClassificationPipeline.fit')
def test_categorical_hyperparam(mock_fit, mock_score, X_y_multi):
```

Reply:
Oh I see. So patching fit/score without calling `mock_fit` or `mock_score` still makes it run faster?